package.json

```diff
@@ -348,6 +377,42 @@
                     "default": false,
                     "markdownDescription": "%config.tableFormatter.normalizeIndentation.description%"
                 },
+                "markdown.extension.theming.decoration.renderCodeSpan": {
```

Comment:
Guess we can directly use `...decoration.codeSpan` and so on

Reply:
The verb "render" **leaves room for** a bunch of other settings that I'm going to add to **this section**.

Besides, I see many people use verb-noun statement as toggle label, and feel the form makes intention clear.

Reply:
I'm okay with that. But I was wondering whether we should have used an object-type configuration, otherwise there are too many trivial options (esp. if there will be more). The disadvantage is it becomes more complex for both us and users, but I guess this is the price to have an advanced configuration 😥.

Reply:
If it is a toggle, then it is supposed to be a toggle.

If it is a leaf, then it is supposed to be a leaf.

IMO, wrapping a group of settings inside an object would introduce an **unnecessary** layer, **harming accessibility, usability, clarity, and validation**.

I don't consider "trivial" items as a problem. Users can benefit a lot from visually and directly editing configuration **right in Settings UI**. And you can see other products also show many "trivial" settings for fine-grained control.

Reply:
https://code.visualstudio.com/api/references/extension-guidelines#settings

https://docs.microsoft.com/en-us/windows/uwp/design/app-settings/guidelines-for-app-settings

https://developer.apple.com/design/human-interface-guidelines/macos/app-architecture/preferences/

https://developer.android.com/guide/topics/ui/settings/organize-your-settings

Reply:
The real problem is **VS Code doesn't provide _setting groups_**. Ideally it should have a tree-like structure, but what we have now is **a big flat list** (visually).

(I have a bad experience when I search "python lint" in the Settings UI: there are 41 results from the Python extension, including different providers: Bandit, Flake8, Mypy, Prospector, Pydocstyle etc., each of which further has several options: enabled, path, args etc. This looks really "noisy" and horrible.)

![image](https://user-images.githubusercontent.com/7588612/107618384-75a68280-6c8c-11eb-8ec9-a01069ef0b6e.png)

I agree that object-type configuration has many disadvantages as you mentioned. To me it is just a workaround and that's why I'm hesitant. If we only have 5 new settings, it won't be a problem. But if each of which will have, e.g. 2 more settings in the future, it becomes more like the Python case 😓.

Reply:
> VS Code doesn't provide setting groups

IMHO, it is VS Code's fault. <https://github.com/microsoft/vscode/issues/70589>

We should not make it worse by introducing custom objects.

---

> I have a bad experience when I search "python lint"

TypeScript's settings are also a beast. 😂

Reply:
> VS Code's fault. microsoft/vscode#70589

Upvoted 👍

> We should not make it worse by introducing custom objects.

Fine, let's just use the simple way (the flat list).